### PR TITLE
This merges upto upstream commit 246893d (tag: REL9_4_24) 

### PR DESCRIPTION
**Pull request by dgkimura**
Notable bug fixes since 9.4STABLE merge (9.4.23)

* Fix daterange canonicalization for +/- infinity.

* Further fix ALTER COLUMN TYPE's handling of indexes and index
constraints.

@pf-qiu - we noticed that upstream commit 53fd0f04b87bf37288f65b5944ac30c64296a745 looks like a modified version of your commit 8b726e8976f in GPDB.  So we were hoping that you could help us figure out what the final version should look like?

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by ashwinstar** [comment]
Do we wish to run full pipeline for this change? As seems spread out changes in various areas.

**Comment by pf-qiu** [comment]
For this conflict in src/tools/msvc/MSBuildProject.pm, please ignore my commit in gpdb repo and just pick the upstream one.

**Comment by dgkimura** [comment]
On Mon, Aug 19, 2019 at 7:50 PM Peifeng Qiu <wrote:
> For this conflict in src/tools/msvc/MSBuildProject.pm, please ignore my commit in gpdb repo and just pick the upstream one.

Thanks for the help.  Pushed commit to resolve accordingly.

**Comment by dgkimura** [comment]
On Mon, Aug 19, 2019 at 5:50 PM Ashwin Agrawal wrote:
> Do we wish to run full pipeline for this change? As seems spread out changes in various areas.

Yup, sounds like good idea.  Pipeline here: https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-merge_REL9_4_24

File: configure [hunk 1]
```diff
@@ -813,6 +813,7 @@ infodir
 docdir
 oldincludedir
 includedir
+runstatedir
```
Comment (d) [inline configure]: this smells like it's generated from a pre-release of `autoconf`, please use the latest version 2.69 of `autoconf`

**Comment by d** [comment]
the autoconf is definitely from a pre-release, I can re-generate those and push a commit

**Comment by dgkimura** [comment]
On Fri, Aug 23, 2019 at 10:44 AM Jesse Zhang wrote:
> the autoconf is definitely from a pre-release, I can re-generate those and push a commit

Thanks Jesse! That's a pretty amazing catch.

**Comment by d** [comment]
> the autoconf is definitely from a pre-release, I can re-generate those and push a commit

done

**Comment by dgkimura** [comment]
Merged https://github.com/greenplum-db/gpdb/commit/526e256b03883bced3aeba70d76d93a717d57bed.  Thanks for the reviews!

**Pull request closed without merging**